code/modules/clothing/suits/marine_armor.dm

```diff
@@ -362,75 +365,99 @@ var/list/squad_colors = list(rgb(230,25,25), rgb(255,195,45), rgb(200,100,200),
 
 	var/list/details = list()
 	var/dose_administered = null
-	var/tricordrazine = REAGENTS_OVERDOSE - (wearer.reagents.get_reagent_amount("tricordrazine") + 0.5)
+	var/tricordrazine = CLAMP(REAGENTS_OVERDOSE - (wearer.reagents.get_reagent_amount("tricordrazine") + 0.5),0,REAGENTS_OVERDOSE - 0.5)
 
 	if(wearer.getFireLoss() > B18_automed_damage && !B18_burn_cooldown)
-		var/dermaline = REAGENTS_OVERDOSE*0.5 - (wearer.reagents.get_reagent_amount("dermaline") + 0.5)
-		var/kelotane = REAGENTS_OVERDOSE - (wearer.reagents.get_reagent_amount("kelotane") + 0.5)
-		if(dermaline < REAGENTS_OVERDOSE*0.5)
-			wearer.reagents.add_reagent("dermaline",REAGENTS_OVERDOSE*0.5 - dermaline + 0.5)
-		if(kelotane < REAGENTS_OVERDOSE)
-			wearer.reagents.add_reagent("kelotane",REAGENTS_OVERDOSE - kelotane + 0.5)
-		if(tricordrazine < REAGENTS_OVERDOSE)
-			wearer.reagents.add_reagent("tricordrazine",REAGENTS_OVERDOSE - tricordrazine + 0.5)
-		details +=("Significant tissue burns detected. Restorative injection administered. <b>Dosage: Dermaline: [dermaline]U | Kelotane: [kelotane]U | Tricordrazine: [tricordrazine]U</b></br>")
-		B18_burn_cooldown = world.time + B18_CHEM_COOLDOWN
-		handle_chem_cooldown(B18_BURN_CODE)
-		dose_administered = TRUE
+		var/dermaline = CLAMP(REAGENTS_OVERDOSE*0.5 - (wearer.reagents.get_reagent_amount("dermaline") + 0.5),0,REAGENTS_OVERDOSE*0.5 - 0.5)
+		var/kelotane = CLAMP(REAGENTS_OVERDOSE - (wearer.reagents.get_reagent_amount("kelotane") + 0.5),0,REAGENTS_OVERDOSE - 0.5)
+		var/burndose = null
+		if(dermaline)
+			wearer.reagents.add_reagent("dermaline",dermaline)
+			burndose = TRUE
+		if(kelotane)
+			wearer.reagents.add_reagent("kelotane",kelotane)
+			burndose = TRUE
+		if(tricordrazine)
+			wearer.reagents.add_reagent("tricordrazine",tricordrazine)
+			burndose = TRUE
+		if(burndose) //Only report if we actually administer something
```

Comment:
You could just do `if(tricordrazine || kelotane || dermaline)` instead of making another var etcetera, so to cut down lenght on the code.
Same applies to other doses.